pyexchange/uniswapv2_analytics.py

```diff
@@ -53,25 +52,25 @@ def from_our_trades_message(trade: dict, pair: str, base_token: Token, our_liqui
         assert (isinstance(base_token, Token))
         assert (isinstance(our_liquidity_balance, Wad))
         assert (isinstance(previous_base_token_reserves, Wad))
+        assert (isinstance(timestamp, int))
 
-        our_pool_share = our_liquidity_balance / Wad.from_number(trade['pair']['totalSupply'])
+        our_pool_share = our_liquidity_balance / Wad.from_number(trade['totalSupply'])
 
-        if trade['pair']['token0']['id'] == base_token.address.address:
+        if trade['token0']['id'] == base_token.address.address:
             swap_price = Wad.from_number(trade['reserve1']) / Wad.from_number(trade['reserve0'])
 
             is_sell = Wad.from_number(trade['reserve1']) < previous_base_token_reserves
 
-            amount = our_pool_share * Wad.from_number(trade['hourlyVolumeToken0'])
+            amount = our_pool_share * abs(Wad.from_number(trade['reserve1']) - previous_base_token_reserves)
```

Comment:
I'm curious why the `abs` was necessary; could a pool have negative tokens, and if it did, why would we pretend the value is positive?

Reply:
Negative reserve's aren't possible with Uniswap or other `x * y = k` exchanges as negative number's would necessarily violate the constant product invariate. 

I was taking the `abs` in order to determine the difference in reserves between the two periods, while avoiding any issues from negative amounts in our calculations. For example, if the previous reserves were greater than i'm assuming we've essentially sold off some amount of the reserves. If less, than we've essentially bought the base currency.